services/wfm/src/main/java/org/openkilda/wfm/topology/event/NetworkTopologyBolt.java

```diff
@@ -0,0 +1,183 @@
+        }
+    }
+
+    private void handleSwitchEvents(SwitchInfoData data) {
+        logger.debug("State update switch {} message {}", data.getSwitchId(), data.getState());
+
+        switch (data.getState()) {
+
+            case ADDED:
+                switchService.createOrUpdateSwitch(SwitchMapper.INSTANCE.map(data));
+                break;
+
+            case ACTIVATED:
+                switchService.activateSwitch(SwitchMapper.INSTANCE.map(data));
+                break;
+
+            case DEACTIVATED:
+            case REMOVED:
+                switchService.deactivateSwitch(SwitchMapper.INSTANCE.map(data));
+                break;
+
+            default:
+                logger.warn("Unknown state update switch info message");
+        }
+    }
+
+    private void handleIslEvents(IslInfoData data, Sender sender) {
+        logger.debug("State update isl {}. Isl state: {}", data.getId(), data.getState());
+
+        switch (data.getState()) {
+            case DISCOVERED:
+                islService.createOrUpdateIsl(IslMapper.INSTANCE.map(data));
+
+                String reason = String.format("Create or update ISL: %s. ISL state: %s",
+                        data.getId(), data.getState());
+                sender.sendRerouteInactiveFlowsMessage(reason);
+                break;
+
+            case FAILED:
+            case MOVED:
+                islService.islDiscoveryFailed(IslMapper.INSTANCE.map(data));
+
+                reason = String.format("ISL discovery failed: %s. ISL state: %s",
+                        data.getId(), data.getState());
+                sender.sendRerouteAffectedFlowsMessage(data.getPath().get(0), reason);
+                break;
+
+            default:
+                logger.warn("Unknown state update isl info message");
+        }
+    }
+
+
+    private void handlePortEvents(PortInfoData data, Sender sender) {
+        logger.debug("State update port {}_{} message cached {}",
+                data.getSwitchId(), data.getPortNo(), data.getState());
+
+        switch (data.getState()) {
+            case DOWN:
+            case DELETE:
+                portService.processWhenPortIsDown(PortMapper.INSTANCE.map(data));
+
+                String reason = String.format("Port %s_%s is %s",
+                        data.getSwitchId(), data.getPortNo(), data.getState());
+                PathNode pathNode = new PathNode(data.getSwitchId(), data.getPortNo(), 0);
+                sender.sendRerouteAffectedFlowsMessage(pathNode, reason);
+                break;
+
+            default:
+                logger.warn("Unknown state update port info message");
+                break;
+        }
+    }
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer declarer) {
+        Fields fields = new Fields(FieldNameBasedTupleToKafkaMapper.BOLT_KEY,
+                FieldNameBasedTupleToKafkaMapper.BOLT_MESSAGE);
+        declarer.declareStream(REROUTE_STREAM, fields);
```

Comment:
You can use "default" (by not passing stream name here) stream name, if bolt have only one output stream.

(minor).